tests/model/api/test_api_generator.py

```diff
@@ -0,0 +1,24 @@
+from unittest import TestCase
+from unittest.mock import Mock, patch
+
+from parameterized import parameterized
+
+from samtranslator.model import InvalidResourceException
+from samtranslator.model.api.api_generator import ApiGenerator
+
+
+class TestApiGenerator(TestCase):
+    @parameterized.expand([("this should be a dict",), ("123",), ([{}],)])
+    @patch("samtranslator.model.api.api_generator.AuthProperties")
```

Comment:
Don't we have some way (functional tests?) to pass a template and ensure it fails some expected manner? Instead of requiring all this mocking.

Reply:
Good call. Added one for the invalid template I intended to fix.